src/clojure/uncomplicate/clojurecl/internal/impl.clj

```diff
@@ -429,12 +429,28 @@
 (extend-mem-array (Class/forName "[S") shorts Short/BYTES)
 (extend-mem-array (Class/forName "[C") chars Character/BYTES)
 
-(extend-type ByteBuffer
+(def component-size-lookup
+  {Float/TYPE     Float/BYTES
+   Double/TYPE    Double/BYTES
+   Long/TYPE      Long/BYTES
+   Integer/TYPE   Integer/BYTES
+   Short/TYPE     Short/BYTES
+   Character/TYPE Character/BYTES
+   Byte/TYPE      Byte/BYTES})
+
+(defn component-size-raw [cls]
+  (get component-size-lookup
+       (.getReturnType (.getMethod cls "get" (into-array Class [])))))
+
+(def component-size (memoize component-size-raw))
+
+(extend-type Buffer
   Mem
   (ptr [this]
     (Pointer/toBuffer this))
   (size [this]
-    (.capacity ^ByteBuffer this)))
+    (* (component-size (type this))
```

Comment:
this is the only real change

Reply:
Did you measure the performance penalty exactly? `size` is used everywhere, so it is very important for it to be as fast as possible.

Reply:
Hmm... memoize is pretty fast, type should be as well, but in that case, the `extend-type` for ByteBuffer should be kept.

Reply:
How many nanoseconds exactly?